### PR TITLE
Tweaks to the EF Core-based health checks

### PR DESCRIPTION
**Pull request by divega**
cc @glennc @rynowak @unaizorrilla @guardrex @Rick-Anderson 

I actually wouldn't mind too much if we want to refer to the DbContext as "database context", but we need to be consistent for the links to work :trollface:

### CODE REVIEW

**Comment by divega** [comment]
Apparently we still need to add this to the ToC for it to show up.

File: aspnetcore/host-and-deploy/health-checks.md [hunk 1]
```diff
@@ -30,7 +30,7 @@ Health checks are usually used with an external monitoring service or container
 
 Reference the [Microsoft.AspNetCore.App metapackage](xref:fundamentals/metapackage-app) or add a package reference to the [Microsoft.AspNetCore.Diagnostics.HealthChecks](https://www.nuget.org/packages/Microsoft.AspNetCore.Diagnostics.HealthChecks) package.
 
-The sample app provides start-up code to demonstrate health checks for several scenarios. The [database probe](#database-probe) scenario probes the health of a database connection using [BeatPulse](https://github.com/Xabaril/BeatPulse). The [database context](#database-context) scenario probes a database context. To explore the database scenarios using the sample app:
+The sample app provides start-up code to demonstrate health checks for several scenarios. The [database probe](#database-probe) scenario probes the health of a database connection using [BeatPulse](https://github.com/Xabaril/BeatPulse). The [`DbContext` probe](#entity-framework-core-dbcontext-probe) scenario probes a database using an EF Core `DbContext`. To explore the database scenarios using the sample app:
```
Comment (guardrex) [inline aspnetcore/host-and-deploy/health-checks.md]: Un-styled link text ...

```suggestion
The sample app provides start-up code to demonstrate health checks for several scenarios. The [database probe](#database-probe) scenario probes the health of a database connection using [BeatPulse](https://github.com/Xabaril/BeatPulse). The [DbContext probe](#entity-framework-core-dbcontext-probe) scenario probes a database using an EF Core `DbContext`. To explore the database scenarios using the sample app:
```

**Comment by tdykstra** [comment]
> It was pulled because I thought @tdykstra was re-working the TOC into yml and the md version was about to be dropped.

Yes I will be doing that, but the yml will be converted from md, so everything should be in md before I start.

**Comment by divega** [comment]
Feedback addressed. Feel free to merge.

**Pull request merged**